executor/runtime/docker/docker.go

```diff
@@ -542,15 +521,34 @@ func netflixLoggerTempDir(cfg config.Config, c *runtimeTypes.Container) string {
 }
 
 func (r *DockerRuntime) setupAdditionalCapabilities(c *runtimeTypes.Container, hostCfg *container.HostConfig) {
+	addedCapabilities := make(map[string]struct{})
+
 	// Set any additional capabilities for this container
 	if cap := c.TitusInfo.GetCapabilities(); cap != nil {
 		for _, add := range cap.GetAdd() {
+			addedCapabilities[add.String()] = struct{}{}
 			hostCfg.CapAdd = append(hostCfg.CapAdd, add.String())
 		}
 		for _, drop := range cap.GetDrop() {
 			hostCfg.CapDrop = append(hostCfg.CapDrop, drop.String())
 		}
 	}
+
+	// Privileged containers automaticaly deactivate seccomp and friends, no need to do this
+	if c.TitusInfo.GetAllowNestedContainers() {
+		hostCfg.SecurityOpt = append(hostCfg.SecurityOpt, "apparmor:docker-nested")
+
+		hostCfg.SecurityOpt = append(hostCfg.SecurityOpt, fmt.Sprintf("seccomp=%s", string(seccomp.MustAsset("nested-container.json"))))
+
+		if _, ok := addedCapabilities["SYS_ADMIN"]; !ok {
+			hostCfg.CapAdd = append(hostCfg.CapAdd, "SYS_ADMIN")
```

Comment:
bold @sargun, bold :clap:

Reply:
lol. I'll unfuck this. I promise. See my TODOs.